schema/handler_test.go

```diff
@@ -61,68 +63,177 @@ func TestHandler(t *testing.T) {
 		return s
 	}
 
-	getFromTS := func(id string, expectCode int) string {
-		res, err := ts.Client().Get(fmt.Sprintf("%s/schemas/%s", ts.URL, id))
+	getFromTS := func(url string, expectCode int) []byte {
+		res, err := ts.Client().Get(url)
 		require.NoError(t, err)
 		body, err := ioutil.ReadAll(res.Body)
 		require.NoError(t, err)
 		require.NoError(t, res.Body.Close())
 
 		require.EqualValues(t, expectCode, res.StatusCode, "%s", body)
-		return string(body)
+		return body
+
 	}
 
-	getFromFS := func(id string) string {
-		f, err := os.Open(strings.TrimPrefix(getSchemaById(id).RawURL, "file://"))
-		require.NoError(t, err)
-		raw, err := ioutil.ReadAll(f)
+	getFromTSById := func(id string, expectCode int) []byte {
+		return getFromTS(fmt.Sprintf("%s/schemas/%s", ts.URL, id), expectCode)
+	}
+
+	getFromTSPaginated := func(page, perPage, expectCode int) []byte {
+		return getFromTS(fmt.Sprintf("%s/schemas?page=%d&per_page=%d", ts.URL, page, perPage), expectCode)
+	}
+
+	getFromFS := func(id string) []byte {
+		raw, err := os.ReadFile(strings.TrimPrefix(getSchemaById(id).RawURL, "file://"))
 		require.NoError(t, err)
-		require.NoError(t, f.Close())
-		return string(raw)
+		return raw
 	}
 
-	var schemasConfig []config.Schema
-	for _, s := range schemas {
-		if s.ID != config.DefaultIdentityTraitsSchemaID {
-			schemasConfig = append(schemasConfig, config.Schema{
-				ID:  s.ID,
-				URL: s.RawURL,
-			})
+	getSchemasConfig := func() (schemasConfig []config.Schema) {
+		for _, s := range schemas {
+			if s.ID != config.DefaultIdentityTraitsSchemaID {
+				schemasConfig = append(schemasConfig, config.Schema{
+					ID:  s.ID,
+					URL: s.RawURL,
+				})
+			}
 		}
+		return schemasConfig
 	}
```

Comment:
Maybe make it less hacky by
```suggestion
	setSchemas := func(newSchemas schema.Schemas) () {
		schemas = newSchemas
		schemasConfig := make([]config.Schema)
		for _, s := range schemas {
			if s.ID != config.DefaultIdentityTraitsSchemaID {
				schemasConfig = append(schemasConfig, config.Schema{
					ID:  s.ID,
					URL: s.RawURL,
				})
			}
		}
		conf.MustSet(config.ViperKeyIdentitySchemas, schemasConfig)
	}
```